crates/biome_js_analyze/src/semantic_analyzers/nursery/no_misleading_character_class.rs

```diff
@@ -0,0 +1,646 @@
+use crate::{semantic_services::Semantic, JsRuleAction};
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, FixKind, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsCallArgument, AnyJsExpression, AnyJsLiteralExpression, AnyJsTemplateElement,
+    JsCallArguments, JsCallExpression, JsNewExpression, JsRegexLiteralExpression,
+    JsStringLiteralExpression, JsSyntaxKind, JsSyntaxToken, T,
+};
+use biome_rowan::{
+    declare_node_union, AstNode, AstNodeList, AstSeparatedList, BatchMutationExt, TextRange,
+};
+declare_rule! {
+    /// Disallow characters which are made with multiple code points in character class syntax
+    ///
+    /// Unicode includes the characters which are made with multiple code points. e.g. Á, 🇯🇵, 👨‍👩‍👦.
+    /// A RegExp character class `/[abc]/` cannot handle characters that consists of multiple code points.
+    /// For example, the character `❇️` consists of two code points: `❇` (U+2747) and `VARIATION SELECTOR-16` (U+FE0F).
+    /// If this character is in a RegExp character class, it will match to either `❇` or `VARIATION SELECTOR-16` rather than `❇️`.
+    /// This rule reports the regular expressions which include multiple code point characters in character class syntax.
```

Comment:
```suggestion
    /// Unicode includes the characters which are made with multiple code points. e.g. Á, 🇯🇵, 👨‍👩‍👦.
    /// A RegExp character class `/[abc]/` cannot handle characters with multiple code points.
    /// For example, the character `❇️` consists of two code points: `❇` (U+2747) and `VARIATION SELECTOR-16` (U+FE0F).
    /// If this character is in a RegExp character class, it will match either `❇` or `VARIATION SELECTOR-16` rather than `❇️`.
    /// This rule reports the regular expressions which include multiple code point characters in character class syntax.
```